### PR TITLE
Feature/send email

### PR DESCRIPTION
**Pull request by AlexTeusz**
Created custom node to send emails with attachments.

Azure PR: 6393

### CODE REVIEW

File: modules/cognigy/src/module.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    const { secret, fromName, fromEmail, to, subject, message, attachmentName, attachmentUrl, contextStore, stopOnError } = args;
+    const { host, port, secure, user, password } = secret;
+
+    // checking arguments
+    if (!secret) throw new Error('No secret defined. You need the secret to proivde the email configuration information.');
+    if (!fromName) throw new Error('No `from` name defined. This could be the name of your company or your employee, for example.');
+    if (!fromEmail) throw new Error('No `from` email address defined.');
+    if (!to) throw new Error('No `to` email address defined. You can provide a list of email addresses by just adding them like this: test@test.de, mail@mail.de, ...');
+    if (!subject) throw new Error('No subject defined.');
+    if (!message) throw new Error('No message defined.');
```
Comment (pedily) [inline modules/cognigy/src/module.ts]: It is possible to send emails without a message

File: modules/cognigy/src/module.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    if (!secure) throw new Error('No email secure option defined. This could be true or false.');
+    if (!user) throw new Error('No email user defined. This is your email username.');
+    if (!password) throw new Error('No email password defined. This is your email password.');
+
+    // validate the given email addresses.
+    const fromMailValidation = validateEmail(fromEmail);
+    if (!fromMailValidation) throw new Error(`The email ${fromEmail} is not valid. Please check it.`);
+
+    for (let em of to.split(',')) {
+        let toMailValidation = validateEmail(em);
```
Comment (pedily) [inline modules/cognigy/src/module.ts]: you should trim `em` before validating it.

File: modules/cognigy/src/module.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    subject: object,
+    message: string
+    attachmentName?: string,
+    attachmentUrl?: string,
+    contextStore: string,
+    stopOnError: boolean
+}): Promise<IFlowInput | {}> {
+
+    const { secret, fromName, fromEmail, to, subject, message, attachmentName, attachmentUrl, contextStore, stopOnError } = args;
+    const { host, port, secure, user, password } = secret;
```
[the remark this answers is not in the document]
Comment (AlexTeusz) [reply modules/cognigy/src/module.ts]: done

File: modules/cognigy/src/module.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    // checking arguments
+    if (!secret) throw new Error('No secret defined. You need the secret to proivde the email configuration information.');
+    if (!fromName) throw new Error('No `from` name defined. This could be the name of your company or your employee, for example.');
+    if (!fromEmail) throw new Error('No `from` email address defined.');
+    if (!to) throw new Error('No `to` email address defined. You can provide a list of email addresses by just adding them like this: test@test.de, mail@mail.de, ...');
+    if (!subject) throw new Error('No subject defined.');
```
[the remark this answers is not in the document]
Comment (AlexTeusz) [reply modules/cognigy/src/module.ts]: done

File: modules/cognigy/src/module.ts (hunk 1 above)
Comment (AlexTeusz) [reply modules/cognigy/src/module.ts]: done

File: modules/cognigy/src/module.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    if (!message) throw new Error('No message defined.');
+
+    // check the attachment argument information
+    if (attachmentName && !attachmentUrl) throw new Error('You have to define botch attachment information. You forgot to define the attachment URL');
```
[the remark this answers is not in the document]
Comment (AlexTeusz) [reply modules/cognigy/src/module.ts]: done

File: modules/cognigy/src/module.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    if (!attachmentName && attachmentUrl) throw new Error('You have to define botch attachment information. You forgot to define the attachment name');
```
[the remark this answers is not in the document]
Comment (AlexTeusz) [reply modules/cognigy/src/module.ts]: done

File: modules/cognigy/README.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        - key: **host**
+        - value: *The STMP host of your Mailserver. Please ask your administrator.*
+    - Email Server Port:
+        - key: **port**
+        - value: The port of your email server; 465 or 587
+    - Email Server Secure Option:
+        - key: **secure**
+        - value: `true` or `false`
+    - Email Username:
+     - key: **username**
```
Comment (pedily) [inline modules/cognigy/README.md]: In the code you check for `user`, in the readme you write `username`. Please update one of them so they match.

File: modules/cognigy/src/module.ts (hunk 2 above)
Comment (pedily) [reply modules/cognigy/src/module.ts]: fixed

File: modules/cognigy/README.md (hunk 7 above)
Comment (pedily) [reply modules/cognigy/README.md]: fixed

**Pull request merged**